patches/ungoogled-chromium/windows/windows-disable-machine-id.patch

```diff
@@ -0,0 +1,53 @@
+# Disable machine ID generation on Windows.
+# See https://github.com/Eloston/ungoogled-chromium/issues/444
+
+--- a/services/preferences/tracked/device_id_win.cc
++++ b/services/preferences/tracked/device_id_win.cc
+@@ -10,11 +10,15 @@
+
+ #include <memory>
+
++#include "base/command_line.h"
+ #include "base/logging.h"
+ #include "base/macros.h"
+
+ MachineIdStatus GetDeterministicMachineSpecificId(std::string* machine_id) {
+   DCHECK(machine_id);
++  if (base::CommandLine::ForCurrentProcess()->HasSwitch("disable-machine-id")) {
++    return MachineIdStatus::NOT_IMPLEMENTED;
++  }
+
+   wchar_t computer_name[MAX_COMPUTERNAME_LENGTH + 1] = {};
+   DWORD computer_name_size = arraysize(computer_name);
+--- a/components/metrics/machine_id_provider_win.cc
++++ b/components/metrics/machine_id_provider_win.cc
+@@ -9,6 +9,7 @@
+ #include <winioctl.h>
+ 
+ #include "base/base_paths.h"
++#include "base/command_line.h"
+ #include "base/files/file_path.h"
+ #include "base/path_service.h"
+ #include "base/threading/scoped_blocking_call.h"
+@@ -18,6 +19,9 @@ namespace metrics {
+ 
+ // static
+ bool MachineIdProvider::HasId() {
++  if (base::CommandLine::ForCurrentProcess()->HasSwitch("disable-machine-id")) {
++    return false;
++  }
+   return true;
+ }
+ 
+@@ -25,6 +29,11 @@ bool MachineIdProvider::HasId() {
+ // is running from.
+ // static
+ std::string MachineIdProvider::GetMachineId() {
++  if (base::CommandLine::ForCurrentProcess()->HasSwitch("disable-machine-id")) {
++    NOTREACHED();
```

Comment:
What is the purpose of `NOTREACHED()` here? Does `disable-machine-id` break debug builds or something?

Reply:
The purpose of `NOTREACHED()` is to raise an assertion (i.e. crash the browser) when the code is reached. So, `NOTREACHED()` in this context doesn't make sense. The reason why this works is because `NOTREACHED()` is implemented as `DCHECK(false)`, which only triggers in debug builds; ungoogled-chromium is built in release mode by default. [See the documentation here for more info](https://chromium.googlesource.com/chromium/src/+/master/styleguide/c++/c++.md#check_dcheck_and-notreached). (Sidenote: The fact that Linux and macOS have `NOTREACHED()` further confirms that these platforms aren't supposed to use the machine ID mechanism.)

I don't see anything other troublesome code here, but please check that the code immediately affected by your changes makes sense in the future. Otherwise, we'd be accumulating technical debt that can be challenging to solve (e.g. #362).

Reply:
You're right, I should have asked about this behavior. But indeed this stub allows to test the code without dealing with this dependency directly.